Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/Sql2016/SqlStorageObjectFactoryTest.cs

```diff
@@ -135,7 +135,7 @@ public void CreateStorageProvider ()
       Assert.That(result.As<RdbmsProvider>().StorageProviderDefinition, Is.SameAs(_rdbmsProviderDefinition));
       Assert.That(result.As<RdbmsProvider>().ConnectionString, Is.SameAs(_rdbmsProviderDefinition.ConnectionString));
 
-      var commandFactory = (RdbmsProviderCommandFactory)PrivateInvoke.GetNonPublicProperty(result, "StorageProviderCommandFactory");
+      var commandFactory = (RdbmsProviderCommandFactory)((RdbmsProvider)result).RdbmsProviderCommandFactory;
```

Comment:
```suggestion
      var commandFactory = (RdbmsProviderCommandFactory)result.As<RdbmsProvider>().RdbmsProviderCommandFactory;
```
For consistency reasons. Still not sure if we should keep the outer cast to RdmbsProvidercommandFactory as a cast operation instead of a call to `As<T>()`

Reply:
I changed it to `As<RdbmsProviderCommandFactory>`